src/common_io.cpp

```diff
@@ -665,98 +546,15 @@ std::string CommonIO::maskString(const std::string &str)
 bool CommonIO::isDigit(const std::string &str)
 {
-    using namespace boost::locale;
-    generator gen;
-
-    // Make system default locale global
-    std::locale loc = gen("");
-    std::locale::global(loc);
-    std::cout.imbue(loc);
-
-    std::string::size_type count = 0;
-    std::string::size_type string_size = str.size();
-
-    std::string character;
-    bool utf_found = false;
-    bool is_digit = false;
-
-    int  char_value =  0;
-    std::string::size_type i = 0;
-
-    if(str.empty())
-    {
-        //std::cout << "Exception (Common::NumberOfChars) string length == 0" << std::endl;
-        return is_digit;
-    }
-
-    using namespace boost::locale;
-    boundary::ssegment_index::iterator p,e;
-    while(1)
-    {
-        if(i == string_size) break;
-        char_value = str[i++];
-
-        // Catch if we've receive multi-byte
-        if(!utf_found)
-        {
-            if((char_value & 0xC0) != 0xC0)
-                utf_found = false; // 1 Byte
-
-            if((char_value & 0xE0) == 0xC0)
-                utf_found = true;  // 2;
-
-            if((char_value & 0xF0) == 0xE0)
-                utf_found = true;  // 3;
-
-            if((char_value & 0xF8) == 0xF0)
-                utf_found = true;  // 4;
-        }
-
-        //charValue = 0x8A;  // Test singleByte extended ascii
-        character += char_value;
-        boundary::ssegment_index index(boundary::word,character.begin(),character.end());
-
-        // Catch All MultiByte UTF-8 Character Sequences.
-        if(!utf_found)
-        {
-            // Test for Ascii Digit
-            if(std::isdigit(str[count]))
-                is_digit = true;
-            else
-            {
-                is_digit = false; // Check for more then one!
-                return is_digit;  // Exit on any failures!
-            }
-            character.erase();
-            ++count;
-        }
-        else
-        {
-            // Parse Character for complete UTF-8 Sequence
-            for(p = index.begin(), e = index.end(); p != e; ++p)
-            {
-                if(p->rule() & boundary::word_none)
-                {
-                    character.erase();
-                    continue;
-                }
-                if(p->rule() & boundary::word_number)
-                {
-                    character.erase();
-                    ++count;
-                    utf_found = false;
-                    is_digit = true;
-                    continue;
-                }
-                else
-                {
-                    is_digit = false;
-                    return is_digit;  // Exit on any failures!
-                }
-            }
-        }
-    }
-    return is_digit;
+    // Later reference for 
+    // Better wide characters.
+    // https://www.cs.helsinki.fi/group/boi2016/doc/cppreference/reference
+    //     /en.cppreference.com/w/cpp/locale/isdigit.html
+    std::string::size_type num_digits = std::count_if(str.begin(), str.end(), 
```

Comment:
With the removal of Boost Locale, we no longer have Unicode digit glyph support, we can only reply on ascii digits for now.  Will look into more later on.